pdcupdater/handlers/depchain/rpms.py

```diff
@@ -0,0 +1,249 @@
+
+There are two handlers here.  They both respond to koji 'tag' events. One of
+them figures out what the build required at *build* time, and updates PDC with
+that. The other figures out what the build requires at *run* time, and updates
+PDC with that.
+
+"""
+
+import logging
+import time
+
+import pdcupdater.handlers
+import pdcupdater.services
+from pdcupdater.utils import (
+    rawhide_tag,
+    tag2release,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+def interesting_tags():
+    """ Returns a list of "interesting tags".
+
+    We have this hacked out for now for just rawhide.
+
+    Eventually, we should query PDC itself to figure out what tags we should be
+    concerned with.
+
+    """
+
+    #releases = bodhi_releases()
+    #stable_tags = [r['stable_tag'] for r in releases]
+    #return stable_tags + [rawhide_tag()]
+    return [rawhide_tag()]
+
+
+class BaseRPMDepChainHandler(pdcupdater.handlers.BaseHandler):
+    """ Abstract base class. """
+
+    def __init__(self, *args, **kwargs):
+        super(BaseRPMDepChainHandler, self).__init__(*args, **kwargs)
+        self.koji_url = self.config['pdcupdater.koji_url']
+
+    @property
+    def topic_suffixes(self):
+        return ['buildsys.tag']
+
+    def can_handle(self, msg):
+        if not msg['topic'].endswith('buildsys.tag'):
+            return False
+
+        # Ignore secondary arches for now
+        if msg['msg']['instance'] != 'primary':
+            log.debug("From %r.  Skipping." % (msg['msg']['instance']))
+            return False
+
+        interesting = interesting_tags()
+        tag = msg['msg']['tag']
+
+        if tag not in interesting:
+            log.debug("%r not in %r.  Skipping."  % (tag, interesting))
+            return False
+
+        return True
+
+    def _yield_pdc_relationships(self, pdc, release_id):
+        for type in self.managed_types:
+            entries = pdc.get_paged(
+                pdc['release-component-relationships']._,
+                from_component_release=release_id,
+                type=type,
+            )
+            for entry in entries:
+                yield entry
+
+    def _yield_koji_relationships(self, pdc, tag):
+
+        release_id, release = tag2release(tag)
+        # TODO -- this tag <-> release agreement is going to break down with modularity.
+
+        pdcupdater.utils.ensure_release_exists(pdc, release_id, release)
+
+        builds = pdcupdater.services.koji_builds_in_tag(self.koji_url, tag)
+
+        for build in builds:
+            parent = {
+                'name': build['name'],
+                'release': {'release_id': release_id},
+            }
+
+            relationships = self.get_koji_relationships_from_build(
+                self.koji_url, build['build_id'])
+            relationships = list(relationships)
+
+            for relationship, child in relationships:
+                child = {'name': child, 'release': {'release_id': release_id}}
```

Comment:
Could you rename this  variable so you're not overwriting the variable produced from the for loop?